### ISSUE TITLE
When ListView is reversed, accessible users cannot reach appbar and bottomNavigationBar

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#52032]
Internal bug: b/150855012

Use case: For a chat app, ListView is reversed to show the latest messages at the bottom. 

Currently when a ListView with vertical axis is reversed, the behavior is:
1. When ListView is at the bottom, users cannot go down to reach input box, but can go up to reach app bar (see internal bug for a video)
2. When ListView is at the top, users can go down to reach input box, but cannot reach app bar (see internal bug for a video)

As a result, users go all the way down to read new messages, but cannot reach input box to reply; when they try to go up and find old messages, they will reach app bar instead (which means old messages are essentially lost).

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text(widget.title)),
      body: ListView.builder(
          reverse: true,
          controller: ScrollController(),
          itemCount: 15,
          itemBuilder: (context, index) {
            return SizedBox(
              height: 50,
              child: Text('This is the #$index chat message'),
            );
          }),
      bottomNavigationBar: Card(
        color: Colors.blue,
        child:
            const SizedBox(height: 50, child: Text('Message input box')),
      ),
    );
  }
}
```

**Comment by mehmetf** [comment]
This is a blocker for customer:money. They currently have to resort to getting exceptions to be released. Please give us an ETA on this whenever possible.

**Comment by jefflim-google** [comment]
Hi, we're in need of an ETA to get the exception approved -- any rough guidance would help!

**Comment by mehmetf** [comment]
@goderbauer @HansMuller Please comment on this as soon as you can.

**Comment by goderbauer** [comment]
I am taking an initial look at this now.

**Comment by goderbauer** [comment]
(Debugging Notes): 

* The problem goes away when `reverse: true` is removed from the ListView. 
* With reverse: true when the ListView is scrolled all the way to the top (no more content to reveal at the top) the semantics tree claims it can only scroll down (fingers move down on the screen). This is incorrect. You can only scroll up (fingers move up on the screen).
* With reverse: false when the ListView is scrolled all the way to the top, the semantics tree claims it can only scroll up (fingers move up on the screen). This is correct.

This makes me believe that we just calculate the available semantics actions incorrectly when reverse is true. This would mean that TalkBack things there is more content to expose and instead of moving the a11y focus out of the ListView it issues a scroll command to expose that content. Since there is no more content, nothing happens.

Looking at the code that calculates available scroll actions now...

**Comment by goderbauer** [comment]
Suspicion confirmed. In https://github.com/flutter/flutter/blob/aad941e3faeeb0c267164f8bde1d4b5449b16667/packages/flutter/lib/src/widgets/scroll_position.dart#L450-L459 we only consider the axis (horizontal/vertical) when calculating the available semantics actions. We should also be considering the growth direction (reverse true/false). The fix looks straight forward, I'll post a PR shortly.

**Comment by goderbauer** [comment]
PR has been posted and is awaiting review. Regarding ETA, I expect that we can submit this today.
